alpha/veneer/semver/semver.go

```diff
@@ -0,0 +1,335 @@
+package semver
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"path"
+	"sort"
+	"strings"
+
+	"github.com/blang/semver/v4"
+	"github.com/operator-framework/operator-registry/alpha/action"
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+// data passed into this module externally
+type Veneer struct {
+	Ref             string
+	SkipPatch       bool
+	ChannelsMajor   bool
+	ChannelsMinor   bool
+	TemplateStrings []string
+}
+
+// IO structs -- BEGIN
+type SemverVeneerBundleEntry struct {
+	Image string `json:"image,omitempty"`
+}
+
+type CandidateBundles struct {
+	Bundles []SemverVeneerBundleEntry `json:"bundles,omitempty"`
+}
+type FastBundles struct {
+	Bundles []SemverVeneerBundleEntry `json:"bundles,omitempty"`
+}
+type StableBundles struct {
+	Bundles []SemverVeneerBundleEntry `json:"bundles,omitempty"`
+}
+
+type SemverVeneer struct {
+	Schema    string           `json:"schema"`
+	Candidate CandidateBundles `json:"candidate,omitempty"`
+	Fast      FastBundles      `json:"fast,omitempty"`
+	Stable    StableBundles    `json:"stable"`
+}
+
+// IO structs -- END
+
+const (
+	CandidateChannelName = "candidate"
+	FastChannelName      = "fast"
+	StableChannelName    = "stable"
+)
+
+// isvd.Channels["stable"] --> quay.io/foo/foo-bundle[0.1.0]
+type ChannelsDataMap map[string]BundlesDataMap  // channel-name --> BundlesDataMap
+type BundlesDataMap map[string][]semver.Version // bundle-name --> []bundle-versions
+// cname --> bname --> [v0, v1, v2, ...]
+
+type _tokenized_bundle_entry struct {
+	_path    string
+	_version semver.Version
+}
```

Comment:
AFAIK, the `_` prefix is not idiomatic, so I'd drop the `_`

identifiers that begin with lowercase letter are unexported (and can therefore only be used in the same package).